internal/rootcoord/create_collection_task.go

```diff
@@ -77,6 +77,45 @@ func (t *createCollectionTask) validate() error {
 	return nil
 }
 
+func defaultValueTypeMatch(schema *schemapb.CollectionSchema) error {
+	for _, fieldSchema := range schema.Fields {
+		if fieldSchema.GetDefaultValue() != nil {
+			switch fieldSchema.GetDefaultValue().Data.(type) {
+			case *schemapb.ValueField_BoolData:
+				if fieldSchema.GetDataType() != schemapb.DataType_Bool {
+					return merr.WrapErrParameterInvalid("DataType_Bool", "not match", "default value type mismatches field schema type")
+				}
+			case *schemapb.ValueField_IntData:
+				if fieldSchema.GetDataType() != schemapb.DataType_Int32 &&
+					fieldSchema.GetDataType() != schemapb.DataType_Int16 &&
```

Comment:
Need to check whether the given int32 value exceeds the value range of field data type (int8, int16)